libtiledbvcf/src/dataset/tiledbvcfdataset.cc

```diff
@@ -764,9 +764,30 @@ void TileDBVCFDataset::load_field_type_maps_v4(
       throw std::runtime_error(
           "Error appending to header for internal allele frequency.");
     }
+    if (bcf_hdr_append(
+            // TODO: do something better than promoting this pointer type;
```

Comment:
Please capture the `bcf_hdr_append` return code in a separate line and test the return code in the if statement.